### PR TITLE
Add opensearch 

### PR DESCRIPTION
**Pull request by FarhanAliRaza**
fixes https://github.com/djangopackages/djangopackages/issues/1174

I have not been able to test it yet because of a Firefox error it may be because of something I missed. 

Screenshot from 2024-07-17 17-59-27

### CODE REVIEW

**Comment by FarhanAliRaza** [comment]
This error only happens with suggestions normal search works.

**Comment by FarhanAliRaza** [comment]
`NS_ERROR_HOST_IS_IP_ADDRESS: Component returned failure code: 0x804b0051 (NS_ERROR_HOST_IS_IP_ADDRESS) [nsIEffectiveTLDService.getKnownPublicSuffix] 2 SearchEngine.sys.mjs:1654
` This is the error when I type with an open search engine. This is a panel opened with `ctrl shift j`

**Comment by jefftriplett** [comment]
@FarhanAliRaza I can look at this tonight. Possibly during our check-in today. 

I'm a little confused by the Component error too.

**Comment by jefftriplett** [comment]
This looks good locally. I'm trying to figure out the browser side.

**Comment by jefftriplett** [comment]
This appears to work great for me! 

Xnapper-2024-07-17-20 44 02

**Comment by jefftriplett** [comment]
Here is one of my settings which is a little hard to see full. I plugged in both the search url and the suggestions url and it lit up my search box where "cms" shows. 

Xnapper-2024-07-17-20 45 14

File: templates/base.html [hunk 1]
```diff
@@ -17,6 +17,7 @@
         <link rel="stylesheet" href="{{ STATIC_URL }}components/audreyr-topbar/dist/jquery.topbar.min.css" type="text/css" />
         <link rel="stylesheet" href="{{ STATIC_URL }}css/oc.css" type="text/css" />
         <link rel="stylesheet" href="{{ STATIC_URL }}css/tailwindcss.min.css" type="text/css" />
+        <link rel="search" type="application/opensearchdescription+xml" href="{% url 'opensearch-description' %}" title="Django Packages">
```
Comment (jefftriplett) [inline templates/base.html:20]: Note to double check this in production.

File: templates/search_description.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  <OutputEncoding>UTF-8</OutputEncoding>
+  <Developer>Django Packages Team</Developer>
+  <Image width="16" height="16" type="image/png">{{ STATIC_URL }}img/favicon.png</Image>
+  <SyndicationRight>open</SyndicationRight>
+  <AdultContent>false</AdultContent>
+  <Url type="application/x-suggestions+json" rel="suggestions" template="http://{{ request.get_host }}{% url 'opensearch-suggestions' %}?q={searchTerms}"/>
+  <Url type="text/html" rel="results" method="get" template="http://{{ request.get_host }}{% url 'search' %}?q={searchTerms}"/>
+  <Url type="application/opensearchdescription+xml" rel="self" template="http://{{ request.get_host }}{% url 'opensearch-description' %}"/>
+
+</OpenSearchDescription>
```
Comment (jefftriplett) [inline templates/search_description.xml:16]: Please remove extra linebreak: 

```suggestion
</OpenSearchDescription>
```

**Review by jefftriplett — APPROVED** [review]
This is good. Great work! 

I think we should do a follow-up issue to add a test, but I'm going to deploy it to test it. Worse case, we disable the meta+link.

**Comment by jefftriplett** [comment]
This is merged to test in production.

**Pull request merged**